### PR TITLE
[WIP] Switch CircleCI template to CircleCI 2.0.

### LINKED ISSUE #52: Convert CircleCI Scaffold From 1.0 to 2.0
The scaffold generated when CircleCI is chosen is a CircleCI 1.0 config file. Can this be updated for CircleCI 2.0? I'm a Developer Evangelist over at CircleCI and I can assist if needed.

Thanks.

### PR DESCRIPTION
**Pull request by felicianotech**
Fixes #52.

This is a first draft at converting the CircleCI template from using CircleCI 1.0 to 2.0

Notes:

1. I haven't looked into optimizing the config yet. Just trying to do a basic conversion.
2. The PHP Docker images are locked to PHP vX.Y, not a specific bug release. We can change that if the project owners decide they specifically want a full vX.Y.Z.
3. I don't know Mustache well so this config may not be proper syntax for Mustache.
4. I haven't run this config nor tested it yet. I plan to soon but others can join in as well. ;)
5. The new config file should be located at ./.circleci/config.yml instead of ./circle.yml. How would I get that to work here?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -252,7 +252,7 @@ The following files are also included unless the `--skip-tests` is used:
 		default: travis
 		options:
 		  - travis
-		  - circle
+		  - circleci
```
Comment (gitlost) [inline README.md]: This shouldn't be changed I think for BC.

Comment (felicianotech) [reply README.md]: BC?

Comment (gitlost) [reply README.md]: Sorry, backward compatibility.

Comment (felicianotech) [reply README.md]: Ah. Makes sense. I'll revert shortly.

**Comment by felicianotech** [comment]
- rebased on latest master
- addressed comment about backwards compatibility
- added some more CircleCI 2.0 config changes to the actual project outside of the new config itself.

**Comment by travislopes** [comment]
Attempting to run this config errors out when trying to install the dependencies as it doesn't have permission to run aptitude. Run commands as `sudo` resolves that.

After fixing that, it errors out with `mysqladmin: command not found` not being found when running the `install-wp-tests.sh` script.

If I add `mysql-server` as a dependency, the command can then be found but cannot connect to the server:

```
mysqladmin: connect to server at '127.0.0.1' failed
error: 'Access denied for user 'ubuntu'@'127.0.0.1' (using password: NO)'
Exited with code 1
```

**Comment by gitlost** [comment]
@travislopes thanks for trying it out, according to https://hub.docker.com/r/circleci/mysql/ the mysql user name is `root`.

**Comment by felicianotech** [comment]
Added a comment addressing the latest comments here.

File: templates/plugin-circle.mustache [hunk 2]
```diff
@@ -1,39 +1,112 @@
...
+      - run:
+          name: "Setup Environment Variables"
+          command: |
+            echo "export PATH=$HOME/.composer/vendor/bin:$PATH" >> $BASH_ENV
+            source /home/circleci/.bashrc
+      - run:
+          name: "Install Dependencies"
+          command: |
+            sudo apt-get update && sudo apt-get install subversion
+            sudo docker-php-ext-install mysqli
```
Comment (travislopes) [inline templates/plugin-circle.mustache]: Installing the MySQLi extension seemed to do the trick. Tests ran perfectly for me with this new config.

Comment (gitlost) [reply templates/plugin-circle.mustache]: Thanks for the feedback @travislopes , stole that from  https://github.com/wp-cli/scaffold-command/pull/118. Not sure about the way it's installing `mysql-client-5.7` (which is missing from circleci docker image), seems kludgey...

**Comment by felicianotech** [comment]
Awesome. Anything else needed from me?

**Comment by gitlost** [comment]
Cool, if you're happy with it @felicianotech then no, will merge!

**Comment by felicianotech** [comment]
Nope, LGTM though there is conflicts now.

**Comment by felicianotech** [comment]
Okay I merged in master to fix the conflicts.

**Comment by schlessera** [comment]
Thanks for the pull-request (and the patience), @felicianotech !

**Pull request merged**